benchmark_runner/common/oc/oc.py

```diff
@@ -745,11 +745,15 @@ def extract_vm_results(self, vm_name: str = '', start_stamp: str = '', end_stamp
             for line in infile:
                 if start_stamp in line:
                     copy = True
-                    results_list.append(line.strip().split(':')[2:])
                     continue
                 elif end_stamp in line:
                     copy = False
                     continue
                 elif copy:
-                    results_list.append(line.strip().split(':')[1:])
+                    # choose only relevant line between start and end stamps
+                    if 'cloud-init' in line:
+                        if vm_name in line:
+                            results_list.append(line.strip().split(':')[2:])
+                        else:
+                            results_list.append(line.strip().split(':')[1:])
```

Comment:
Could you provide (as a comment) an example of what you're looking to catch with this?  Everything else looks good.

Reply:
Done

Reply:
Could you provide an actual example?  The comment isn't really very helpful.

Reply:
That way, if the output changes later, it will be possible to reconstruct what you were parsing so it will be easier to fix.

Reply:
Here is a vm output example:
```
[   17.785757] cloud-init[1270]: @@~@@START-WORKLOAD@@~@@

CentOS Stream 8
Kernel 4.18.0-365.el8.x86_64 on an x86_64

Activate the web console with: systemctl enable --now cockpit.socket

vdbench-vm-7fcdf173 login: [  300.150058] cloud-init[1270]: Run,Xfersize,Threads,Reqrate,Rate,Rate_std,Rate_max,Resp,Resp_std,Resp_max,MB/sec,MB_read,MB_write,Read_rate,Read_rate_std,Read_rate_max,Read_resp,Read_resp_std,Read_resp_max,Write_rate,Write_rate_std,Write_rate_max,Write_resp,Write_resp_std,Write_resp_max,Mkdir_rate,Mkdir_rate_std,Mkdir_rate_max,Mkdir_resp,Mkdir_resp_std,Mkdir_resp_max,Rmdir_rate,Rmdir_rate_std,Rmdir_rate_max,Rmdir_resp,Rmdir_resp_std,Rmdir_resp_max,Create_rate,Create_rate_std,Create_rate_max,Create_resp,Create_resp_std,Create_resp_max,Open_rate,Open_rate_std,Open_rate_max,Open_resp,Open_resp_std,Open_resp_max,Close_rate,Close_rate_std,Close_rate_max,Close_resp,Close_resp_std,Close_resp_max,Delete_rate,Delete_rate_std,Delete_rate_max,Delete_resp,Delete_resp_std,Delete_resp_max,Getattr_rate,Getattr_rate_std,Getattr_rate_max,Getattr_resp,Getattr_resp_std,Getattr_resp_max,Setattr_rate,Setattr_rate_std,Setattr_rate_max,Setattr_resp,Setattr_resp_std,Setattr_resp_max,Access_rate,Access_rate_std,Access_rate_max,Access_resp,Access_resp_std,Access_resp_max,Compratio,Dedupratio,cpu_used,cpu_user,cpu_kernel,cpu_wait,cpu_idle
[  300.153581] cloud-init[1270]: oltp1,4096,3,9999988,1146.8,112.103,542,7.3616,180.7214,4634.901,4.4797,2.2211,2.2586,568.6,30.6241,581,6.6303,3.7049,1329.045,578.2,30.4439,579,8.0808,3.3257,1332.234,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.16,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,n/a,3.9919,1.7779,2.214,0,0
[  300.154360] cloud-init[1270]: 64kb_write_1threads,65536,1,9999988,298.2667,0,0,3.6945,0,0,18.6417,0,18.6417,0,0,0,0,0,0,298.2667,24.1005,305,3.6945,24.4081,1634.962,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1.8667,1.5275,2,0.1047,0.0457,0.31,1.8667,1.5275,2,0.0212,0.0092,0.056,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,n/a,3.1104,1.8729,1.2375,0,0
[  300.155113] cloud-init[1270]: @@~@@END-WORKLOAD@@~@@
```

Reply:
What happens if the VM name itself contains "cloud_init"?

Are you looking to parse a line that literally contains `@@~@@START-WORKLOAD@@~@@`, or will that be substituted already?  If it literally contains that string, can you search for that?

Reply:
The vm name cannot contains cloud-init because its a constant template with unique id that control by benchmark-runner.

Reply:
This is the way for control several workloads remotely for now. maybe we can improve it in the near future using redis database.

Reply:
I'm suggesting another way of parsing out the timestamp, and want you to add a comment describing precisely what you're searching for.

Reply:
I am not parsing the timestamp at all. I am extracting the output data.

Reply:
So that's why it needs a comment -- its purpose is not evident.  A better comment would be

```
# Strip line prefix from VM output
```

Do that and I'll lgtm.

Reply:
Update